plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/related-by-control.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	CheckboxControl,
+	PanelBody,
+	// @ts-expect-error missing types.
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { QueryControlProps, RelatedBy, CoreFilterNames } from '../../types';
+
+const RelatedByControl = ( {
+	query,
+	setQueryAttribute,
+	trackInteraction,
+}: QueryControlProps ) => {
+	const relatedBy = query.relatedBy as RelatedBy;
```

Comment:
Completely nitpick, but since this is an [optional prop](https://github.com/woocommerce/woocommerce/pull/52580/files#diff-2730289cd10e5059fa0c04f43f2d0c358137074d357d898eccdfbb7fd0e3d890R120) it would make sense to add an optional chaining operator here (`query?.relatedBy`).